### PR TITLE
{ai}[foss/2021a] DGL v0.9.1 w/ Python 3.9.5

### PR DESCRIPTION
**Pull request by akesandgren**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by akesandgren** [comment]
Test report by @akesandgren
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in total)
b-an03.hpc2n.umu.se - Linux Ubuntu 20.04, x86_64, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, 1 x NVIDIA Quadro K6000, 470.161.03, Python 3.8.10
See https://gist.github.com/dfe6cac9554f53ea44088f0268976a41 for a full test report.

**Review by Micket — APPROVED** [review]
lgtm

what a messy setup this reuired, yikes

**Comment by Micket** [comment]
Test report by @Micket
**FAILED**
Build succeeded for 5 out of 7 (5 easyconfigs in total)
alvis1-07 - Linux Rocky Linux 8.6, x86_64, Intel(R) Xeon(R) Gold 6244 CPU @ 3.60GHz, 2 x NVIDIA Tesla V100-SXM2-32GB, 520.61.05, Python 3.6.8
See https://gist.github.com/6517b9446816effd1fdf72d1b721fe29 for a full test report.

File: easybuild/easyconfigs/d/DGL/DGL-0.9.1-foss-2021a-CUDA-11.3.1.eb [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    },
+]
+patches = [
+    '%(name)s-%(version)s_use_externals_instead_of_submodules.patch',
+]
+checksums = [
+    {'dgl-0.9.1.tar.gz': '8d26ebb7ed976665bbf5bbd1792d8e6efb13a8fa16e5eb1efed75e07fb982e04'},
+    {'metis-5.1.1-DistDGL-v0.5.tar.gz': 'cedf0b32d32a8496bac7eb078b2b8260fb00ddb8d50c27e4082968a01bc33331'},
+    {'GKlib-METIS-v5.1.1-DistDGL-0.5.tar.gz': '52aa0d383d42360f4faa0ae9537ba2ca348eeab4db5f2dfd6343192d0ff4b833'},
+    {'tensorpipe-20230206.tar.gz': 'a4246afac190ffc1bef438920cea490a9aa8dfc5983e765b420ec4408331618e'},
```
Comment (Micket) [inline easybuild/easyconfigs/d/DGL/DGL-0.9.1-foss-2021a-CUDA-11.3.1.eb:65]: checksum can't be used here. Pretty sure this dict syntax doesn't allow for `None`  values either last i tried. So, probably need a framework fix for that, ugh

Comment (akesandgren) [reply easybuild/easyconfigs/d/DGL/DGL-0.9.1-foss-2021a-CUDA-11.3.1.eb:65]: ah forgot about that one...

**Comment by Micket** [comment]
nanoflann tests are failing for me
```
[ RUN      ] kdtree.L2_vs_L2_simple
/local/tmp.808931/nanoflann/1.4.0/GCCcore-10.3.0/nanoflann-1.4.0/tests/test_main.cpp:96: Failure
Expected equality of these values:
  out_dist_sqr1[i]
    Which is: 4.5514000000000001
  out_dist_sqr[i]
    Which is: 1.4525000000000001
```
a bunch of these. The errors aren't close. Same on icelake and skylake.

**Comment by akesandgren** [comment]
Hmm I only used broadwell for building this so far...

**Comment by akesandgren** [comment]
Using GCC 11.3.0 instead of 10.3.0 makes the tests pass on skylake for me.
So this might be a compiler bug in 10.3.0...

**Comment by akesandgren** [comment]
And turning off vectoriser for 10.3.0 also makes it pass. so very likely a compiler problem/bug

**Comment by akesandgren** [comment]
Test report by @akesandgren
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in total)
b-cn1501.hpc2n.umu.se - Linux Ubuntu 20.04, x86_64, Intel(R) Xeon(R) Gold 6132 CPU @ 2.60GHz, 2 x NVIDIA Tesla V100-PCIE-16GB, 470.161.03, Python 3.8.10
See https://gist.github.com/99d643a6001d8dd8d4778c0f253a071b for a full test report.

**Comment by Micket** [comment]
Test report by @Micket
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in total)
alvis-c1 - Linux Rocky Linux 8.6, x86_64, Intel Xeon Processor (Skylake), Python 3.6.8
See https://gist.github.com/a477b923803cc891058e1c852bd93c75 for a full test report.

**Comment by Micket** [comment]
Test report by @Micket
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in total)
alvis-s1 - Linux Rocky Linux 8.6, x86_64, Intel(R) Xeon(R) Silver 4314 CPU @ 2.40GHz, Python 3.6.8
See https://gist.github.com/6e6919511d7f82a29b12a258cd83f5be for a full test report.

**Comment by Micket** [comment]
Test report by @Micket
**SUCCESS**
Build succeeded for 2 out of 2 (1 easyconfigs in total)
alvis3-18 - Linux Rocky Linux 8.6, x86_64, Intel(R) Xeon(R) Gold 6338 CPU @ 2.00GHz, 4 x NVIDIA NVIDIA A100-SXM4-40GB, 520.61.05, Python 3.6.8
See https://gist.github.com/75c4cf9843fd3757dfa1efe7354bde2a for a full test report.

**Review by Micket — APPROVED** [review]
lgtm

**Comment by Micket** [comment]
Going in, thanks @akesandgren!

**Pull request merged**